2024-comments-are-a-programmers-love-letters

dujue55 opened the pull request:
Add the first article : Comments are a programmer's love letters

Ajorian commented:
I think the expression 'the secret language of code' is somehow misleading. On the other hand, the negative side of over-commented codes is not considered. Maybe you can talk about a good balance of using comments.

evos96 commented:
External feedback:

negative aspect(s) pointed out:
-I didn't understand the purpose (2)
-The structure was confusing. (1)

positive aspect(s) pointed out:
-The information was easy to understand. (1)
-Good balance between technical depth and readability. (4)
-The structure is clear and easy to follow. (1)
-I understood the purpose (1)

Did the drawing help you to understand the meaning of the article?
yes (5) %71
no (2)

Did you notice any factual errors or inaccuracies in the article? If Yes, which part?
-No it was very easy to understand
-To my limited knowledge everything is up to code

Additional comments
-Nice
-The comment example during the text does not match the content of the text.

carinaschrenk commented:
# Internal Feedback Round 1B 14.11 

## Thoughts 
- Comment is not in line with the context.
- Too long.

## Ideas

dujue55 commented:
Based on the feedback, I made the following changes to the article:

1. Removed the comment example as it was irrelevant and redundant.
2. Added a discussion on balance in using comments.
3. Shortened the overall content.

ParthS007 commented:
## Review Group 5 (Arita, Ivan, Parth)

The revision effectively balances clarity and purpose in discussing the use of comments.

belle903 commented:
**Ideas and notes from the creative writing session today:**

**Ideas regarding examples to visualise also for non-CS people:** 
- sticky notes on the fridge
- notes on walls from ancestors
- sticky notes on coffee machine
    - "don't touch it"
    - "fill beans here”
--> show comments as notes explaining what is in a class/method

**General notes:**
- remove love letter
- comments are not necessary clear and with care
--> as comments are a necessary thing to add to your code, it does not really depend on "love" for another programmer. 

**Suggestions for the title - make it catchy with a concrete example to refer to in the article:**
    - "What does it do?” (like a comment for a method on what the method does)

carinaschrenk commented:
#### As Discussed in the seminar today, here is a sample text for one of the ideas mentioned by Laura:

Do you know how your coffee machine works? Probably not. But you can still make coffee with it. You can even adjust the amount of water or the grind level.

Often, programmers use code without knowing how it works. How can they do this? 
Because the person who wrote the code left comments, explaining what the different parts do. 
You can imagine it like a coffee machine in your office where someone put sticky notes next to the important buttons, explaining for each button what it controls. 
There might even be a sticky note that says: „Do NOT touch!"

selaudin commented:
## Final Feedback from the Content and Quality Team 

1. Remove the last sentence because it addresses the programers.
2. Replace the love letter metaphor with a sticky note or some textbook scribble metaphor.  
3. Make the article inclusive for everyone. We should speak to everyone in the booklet. Like "so add comments only where necessary" could be rephrased or removed.
5. Simplify, rephrase or remove this section "I also think this sentence complicates the article for the reader. Many details makes the article also harder to attract the readers attention. "Still, balance is important: excessive commenting can clutter the code and reduce its readability, so add comments only where necessary.""

**Please revise the article by the 31st**, since the final versions should be available then, as decided in the seminar.

dujue55 commented:
Thank you for your feedback! I’ve revised the article:

1. I kept the “love letter” metaphor because it highlights the personal effort and care programmers put into making their code understandable.

2.I adjusted the wording so it speaks to anyone who might read the comments, rather than focusing only on programmers.

ashdriod commented:
Hi, this is the second delivery of the graphics, which currently include watermarks that will be removed after the delivery is approved on Fiverr. Small changes can be made until Tuesday, while larger changes will be included in the final order.
![26](https://github.com/user-attachments/assets/3a6b0ab1-99dc-4c6b-9bcc-5ecf19741633)

lauraengist commented:
**Edited text after proofreading** (Testing TF):

Well done! We have nothing to edit here.
